File: frontend/src/app/dto/qwirkle/GameState.ts
import {StackTile} from "./StackTile";
import {BoardTile} from "./BoardTile";
import {Tile} from "./Tile";
import {PositionInfo} from "./PositionInfo";

export interface GameState {
    hand: Tile[],
    stack: StackTile[],
    board: BoardTile[],
    openPositions: PositionInfo[]
}

File: backend/src/main/java/net/zorphy/backend/site/qwirkle/dto/GameState.java
package net.zorphy.backend.site.qwirkle.dto;

import net.zorphy.backend.site.qwirkle.dto.tile.BoardTile;
import net.zorphy.backend.site.qwirkle.dto.tile.StackTile;
import net.zorphy.backend.site.qwirkle.dto.tile.Tile;

import java.util.List;

public record GameState(
        List<Tile> hand,
        List<StackTile> stack,
        List<BoardTile> board,
        List<PositionInfo> openPositions
) {
}
